SofaKernel/modules/SofaDeformable/StiffSpringForceField.inl

```diff
@@ -48,16 +48,72 @@ StiffSpringForceField<DataTypes>::StiffSpringForceField(double ks, double kd)
 template<class DataTypes>
 StiffSpringForceField<DataTypes>::StiffSpringForceField(MechanicalState* object1, MechanicalState* object2, double ks, double kd)
     : SpringForceField<DataTypes>(object1, object2, ks, kd)
+    , d_indices1(initData(&d_indices1, "indices1", "Indices of the source points on the first model"))
+    , d_indices2(initData(&d_indices2, "indices2", "Indices of the fixed points on the second model"))
+    , d_length(initData(&d_length, 0.0, "length", "uniform length of all springs"))
 {
 }
 
 
 template<class DataTypes>
 void StiffSpringForceField<DataTypes>::init()
 {
+    if (d_indices1.isSet() && d_indices2.isSet())
+    {        
+        m_dataTrackerIndices.trackData(d_indices1);
+        m_dataTrackerIndices.trackData(d_indices2);
+
+        createSpringsFromInputs();
+    }
+
     this->SpringForceField<DataTypes>::init();
 }
 
+template<class DataTypes>
+void StiffSpringForceField<DataTypes>::updateSpringsIfChanged()
+{
+    if (!d_indices1.isSet() && !d_indices2.isSet()) // nothing to do in this case
+        return;
+
+    // force update
+    d_indices1.updateIfDirty();
+    d_indices2.updateIfDirty();
```

Comment:
this will force the update (if dirty),
means: 
1- it will check if this Data is dirty. This can be the case if a parent Data has been modified. In this case all children tree is set to dirty (but not yet updated)
2- Call the update method, for a simple Data link it means copy new values from parent Data.
3- Set this Data to clean (dirty == false)
4- (this I'm not 100% sure, would need to check the code) Increase the counter of your Data.

5- In the line below m_dataTrackerIndices.hasChanged() we check if all Data counter are still the same as the value stored by the dataTracker. if one is different, hasChanged return true.